container-chains/templates/simple/runtime/src/xcm_config.rs

```diff
@@ -243,3 +253,185 @@ impl cumulus_pallet_dmp_queue::Config for Runtime {
+    type AssetAccountDeposit = ForeignAssetsAssetAccountDeposit;
+    type RemoveItemsLimit = frame_support::traits::ConstU32<1000>;
+    #[cfg(feature = "runtime-benchmarks")]
+    type BenchmarkHelper = ForeignAssetBenchmarkHelper;
+}
+
+impl pallet_foreign_asset_creator::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type ForeignAsset = MultiLocation;
+    type ForeignAssetCreatorOrigin = EnsureRoot<AccountId>;
+    type ForeignAssetModifierOrigin = EnsureRoot<AccountId>;
+    type ForeignAssetDestroyerOrigin = EnsureRoot<AccountId>;
+    type Fungibles = ForeignAssets;
+    type WeightInfo = pallet_foreign_asset_creator::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_asset_rate::Config for Runtime {
+    type CreateOrigin = EnsureRoot<AccountId>;
+    type RemoveOrigin = EnsureRoot<AccountId>;
+    type UpdateOrigin = EnsureRoot<AccountId>;
+    type Currency = Balances;
+    type AssetKind = AssetId;
+    type RuntimeEvent = RuntimeEvent;
+    type WeightInfo = pallet_asset_rate::weights::SubstrateWeight<Runtime>;
+    #[cfg(feature = "runtime-benchmarks")]
+    type BenchmarkHelper = ForeignAssetBenchmarkHelper;
+}
+
+use {
+    crate::ForeignAssets,
+    sp_runtime::{traits::CheckedDiv, FixedPointNumber},
+    staging_xcm_builder::{FungiblesAdapter, NoChecking},
+    staging_xcm_executor::traits::JustTry,
+};
+
+/// Means for transacting foreign assets from different global consensus.
+pub type ForeignFungiblesTransactor = FungiblesAdapter<
+    // Use this fungibles implementation:
+    ForeignAssets,
+    // Use this currency when it is a fungible asset matching the given location or name:
+    (ConvertedConcreteId<AssetId, Balance, ForeignAssetsCreator, JustTry>,),
+    // Convert an XCM MultiLocation into a local account id:
+    LocationToAccountId,
+    // Our chain's account ID type (we can't get away without mentioning it explicitly):
+    AccountId,
+    // We dont need to check teleports here.
+    NoChecking,
+    // The account to use for tracking teleports.
+    CheckingAccount,
+>;
+
+/// Multiplier used for dedicated `TakeFirstAssetTrader` with `ForeignAssets` instance.
+pub type AssetRateAsMultiplier =
+    parachains_common::xcm_config::AssetFeeAsExistentialDepositMultiplier<
+        Runtime,
+        WeightToFee,
+        CustomConverter,
+        ForeignAssetsInstance,
+    >;
+
+// TODO: move to https://github.com/paritytech/polkadot-sdk/pull/2903 once its merged
+pub struct CustomConverter;
+impl frame_support::traits::tokens::ConversionToAssetBalance<Balance, AssetId, Balance>
+    for CustomConverter
+{
+    type Error = ();
+    fn to_asset_balance(balance: Balance, asset_id: AssetId) -> Result<Balance, Self::Error> {
+        let rate = pallet_asset_rate::ConversionRateToNative::<Runtime>::get(asset_id).ok_or(())?;
+        Ok(sp_runtime::FixedU128::from_u32(1)
+            .checked_div(&rate)
+            .ok_or(())?
+            .saturating_mul_int(balance))
+    }
+}
+
```

Comment:
This is going to dissappear once I convince parity to accept https://github.com/paritytech/polkadot-sdk/pull/3318

Hopefully this happens fast